lib/wibox/widget/slider.lua

```diff
@@ -460,25 +462,65 @@ function slider:draw(_, cr, width, height)
     if margins then
         if type(margins) == "number" then
             x_offset, y_offset = margins, margins
-            handle_width  = handle_width  - 2*margins
-            handle_height = handle_height - 2*margins
+            handle_width       = handle_width - 2 * margins
+            handle_height      = handle_height - 2 * margins
         else
             x_offset, y_offset = margins.left or 0, margins.top or 0
-            handle_width  = handle_width  -
-                (margins.left or 0) - (margins.right  or 0)
-            handle_height = handle_height -
-                (margins.top  or 0) - (margins.bottom or 0)
+            handle_width       = handle_width -
+                (margins.left or 0) - (margins.right or 0)
+            handle_height      = handle_height -
+                (margins.top or 0) - (margins.bottom or 0)
         end
     end
 
     -- Get the widget size back to it's non-transfored value
     local min, _, interval = get_extremums(self)
-    local rel_value = math.floor(((value-min)/interval) * (width-handle_width))
+    local rel_value = math.floor(((value - min) / interval) * (width - handle_width))
 
     cr:translate(x_offset + rel_value, y_offset)
 
     handle_shape(cr, handle_width, handle_height)
 
+    local handle_icon = self._private.handle_icon
+        or nil
+
+
+    if handle_icon then
+        cr:save()
+        -- set clip
+        -- cr:clip(self._private.handle_shape(cr, handle_height, handle_width))
+        --cr:translate(230, 230)
+        cr:scale(0.25, 0.25)
+        --cr:translate(handle_width*-0.5, handle_width*-0.5)
+        if type(handle_icon) == "function" then
+            handle_icon(context, cr, handle_width, handle_height, handle_icon_args)
+        else
+            local icon = surface.load(handle_icon)
+            cr:set_source_surface(icon, 0, 0)
+            cr:paint()
+        end
+        cr:restore()
+    end
+
+    --- The background image to use.
+    --
+    -- If `image` is a function, it will be called with `(context, cr, width, height)`
+    -- as arguments. Any other arguments passed to this method will be appended.
+    --
+    -- @property bgimage
+    -- @tparam[opt=nil] image|nil bgimage
+    -- @see gears.surface
+
+    --[[ function slider:set_handle_icon(icon, ...)
+        self._private.handle_icon = type(icon) == "function" and icon or surface.load(icon)
+        self._private.handle_icon_args = { ... }
+        self:emit_signal("widget::redraw_needed")
+        self:emit_signal("property::handle_icon", icon)
+    end ]]
```

Comment:
Code example doesn't need to be wrapped by block style comments. What's important is to indent them further, so ldoc recognizes them as code-block.

By convention, all ldoc block comments are still single line comments starting with `--`.

Alternatively (and preferably for this kind of feature), we write examples in the `test/examples` directory, then reference the file with a `--@DOC_path_with_underscore_EXAMPLE@` comment line. Running the `ldoc` make target (also ran by the default target) generates images from the examples and inserts them at the corresponding comments' location.